src/OrchardCore.Modules/Orchard.Settings/Services/SiteService.cs

```diff
@@ -92,12 +97,18 @@ public class SiteService : ISiteService
             existing.TimeZone = site.TimeZone;
             existing.UseCdn = site.UseCdn;
 
-            _session.Save(existing);
+            session.Save(existing);
 
             _memoryCache.Set(SiteCacheKey, site);
             _signal.SignalToken(SiteCacheKey);
 
             return;
         }
+
+        private YesSql.ISession GetSession()
+        {
+            var httpContextAccessor = _serviceProvider.GetService<IHttpContextAccessor>();
```

Comment:
cant you just inject in IHttpContextAccessor?

Reply:
instead of IServiceScope

Reply:
The reasoning is that it might be more expensive to resolve IHttpContextAccesssor and the main use case is to read the settings which are cached so we will almost never need to resolve it.